### PR TITLE
Takeaways from The Pragmatic Programmer

### PR DESCRIPTION
**Pull request by Dunklas**

### CODE REVIEW

File: content/posts/2022-10-19_learnings-from-pragmatic-programmer-1.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# Broken Window
+The "Broken Window" theory was introduced by James Q. Wilson and George L. Kelling in their article "Broken Windows" fron 1982.
+They argued that small signs of disorder and neglect in a neighborhood, such as broken windows or graffiti, can lead to more serious crime and social problems.
+The main idea is that if a broken window is left unrepaired, it sends a message that no one cares about the area.
+
+In the context of software development, the "Broken Window" theory can be applied to the idea that small problems in our code can have a similar cascading effect. 
+If we allow small issues, such as code that is difficult to read or maintain, to go unaddressed, it can lead to larger problems in the future.
+These larger problems can include increased difficulty in making changes to the code, a higher risk of introducing bugs, and decreased overall quality and reliability of the code.
+
+There is really only one defense against the consequences of broken windows, and that is to fix the broken windows as soon as they are identified.
```
Comment (Dunklas) [inline content/posts/2022-10-19_learnings-from-pragmatic-programmer-1.md:70]: Maybe we should state that broken windows doesn't necessarily need to be _repaired_ immediately, but it's important to at least board them up :)

Comment (mile95) [reply content/posts/2022-10-19_learnings-from-pragmatic-programmer-1.md:70]: Löser!!

File: content/posts/2023-01-19_learnings-from-the-pragmatic-programmer-1.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+In some projects, it may be argued that there's no time to write tests because feature development needs to be prioritized.
+If we accept the fact that testing drives good design, and that good design makes us deliver faster in the long run; you should find a way to write tests anyway, preferably in parallel with the feature development.
+[Friday Facts #366 - The only way to go fast, is to go well!](https://factorio.com/blog/post/fff-366) is a nice blog post about the relationship between software design and delivery speed.
+
+# Easy to Change
+As software developers, we often find ourselves working on projects that require frequent updates and changes.
+Whether it's adding new features, fixing bugs, or adapting to changing requirements, the ability to make changes to our codebase quickly and easily is critical.
+That's where the concept of an "Easy to Change" (ETC) mindset comes in.
+
+The ETC mindset is all about designing software systems that are easy to modify and adapt as needed. This involves writing clean, readable, and well-structured coded while
```
Comment (Dunklas) [inline content/posts/2023-01-19_learnings-from-the-pragmatic-programmer-1.md:56]: coded > code

File: content/posts/2023-01-19_learnings-from-the-pragmatic-programmer-1.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+Another important point to understand is to know when to stop, i.e. having a clear definition of what _good enough_ means for a particular project. Do not break a perfectly good program by spending too much time on overembellishment and overrefinement. There will come a time when you have to move on, and let the code be as it is. It will likely not be perfect, but then again, it can never be.
+
+# Don’t Outrun your Headlights
+
+Think of this scenario. You have worked on a feature for several weeks, everything is set up, the API is finished, you have put the final touches on the UI and it's time to show the customer/stakeholders what you have created. You're full of confidence, you have made your feature according to the designs and specifications provided. But the demo doesn't go well, the stakeholders want to change things or they are not happy with the feature and you slowly realise that most of the work that you have done for the last weeks have been for nothing. The API needs to be redesign, the UI radically altered and the lines of code that needs to be removed are numbered in the hundreds.
```
Comment (Dunklas) [inline content/posts/2023-01-19_learnings-from-the-pragmatic-programmer-1.md:95]: redesign > redesigned

**Pull request merged**